lib/stack_master/stack_dependency.rb

```diff
@@ -0,0 +1,65 @@
+module StackMaster
+  class StackDependency
+    StackOutputNotFound = Class.new(StandardError)
+
+    def initialize(stack_definition, config)
+      @stack_definition = stack_definition
+      @config = config
+    end
+
+    def outdated_stacks
+      @config.stacks.select do |stack|
+        dependent_stack = Stack.find(stack.region, stack.stack_name)
+        next unless dependent_stack
+        parameters = ParameterLoader.load(stack.parameter_files)
+        any_stack_output_outdated?(parameters, dependent_stack) || any_stack_outputs_outdated?(parameters, dependent_stack)
+      end
+    end
+
+    private
+
+    def any_stack_output_outdated?(params, stack)
+      params.any? do |_, value|
+        value['stack_output'] &&
+          value['stack_output'].gsub('_', '-') =~ %r(#{@stack_definition.stack_name}/) &&
+          outdated?(stack, value['stack_output'].split('/').last)
+      end
+    end
+
+    def any_stack_outputs_outdated?(params, stack)
+      params.any? do |key, value|
+        value['stack_outputs'] &&
```

Comment:
Consider checking `stack_outputs` in `any_stack_output_outdated?` to avoid loping through the same `params` twice. It will also simplify the `outdated_stacks` method.